src/provider/provider_tracking.c

```diff
@@ -168,8 +173,8 @@ static umf_result_t trackingAllocationSplit(void *hProvider, void *ptr,
 
     // We'll have a duplicate entry for the range [highPtr, highValue->size] but this is fine,
     // the value is the same anyway and we forbid removing that range concurrently
-    ret = umfMemoryTrackerAdd(provider->hTracker, provider->pool, highPtr,
-                              secondSize);
+    ret = umfMemoryTrackerAdd(provider->hTracker, provider->pool,
+                              provider->ba_pool, highPtr, secondSize);
     if (ret != UMF_RESULT_SUCCESS) {
         fprintf(stderr, "tracking split: umfMemoryTrackerAdd failed\n");
         // TODO: what now? should we rollback the split? This can only happen due to ENOMEM
```

Comment:
It would be great if we could also solve this issue by using base_alloc. I was thinking that we could have a per-thread `ba_pool` and a function like `ba_pool_reserve(num_allocations)`  which we could call at the beginning of this function and it would guarantee there's enough memory to satisfy `num_allocations`. Since the pool would be per-thread, we wouldn't need to worry that someone else would allocate this memory (if other thread frees memory back to the pool, it's fine).

Reply:
I agree it should be done in a separate PR.